satpy/writers/cf_writer.py

```diff
@@ -855,6 +858,11 @@ def save_datasets(self, datasets, filename=None, groups=None, header_attrs=None,
                 grp_str = ' of group {}'.format(group_name) if group_name is not None else ''
                 logger.warning('No time dimension in datasets{}, skipping time bounds creation.'.format(grp_str))
 
+            if group_attrs is not None:
+                if flatten_attrs:
+                    group_attrs = flatten_dict(group_attrs)
+                dataset.attrs = encode_attrs_nc(group_attrs)
+
             encoding, other_to_netcdf_kwargs = update_encoding(dataset, to_netcdf_kwargs, numeric_name_prefix)
```

Comment:
If you put this chunk of code (everything inside the for loop besides the `to_netcdf` and `append` at the end) in a separate method, how ugly would it be?

Reply:
Thanks, that passes now.